internal/dispatch/remote/cluster.go

```diff
@@ -216,6 +216,147 @@ func dispatchRequest[Q requestMessage, S responseMessage](ctx context.Context, c
 	return *new(S), foundError
 }
 
+type requestMessageWithCursor interface {
+	requestMessage
+	GetOptionalCursor() *v1.Cursor
+}
+
+type responseMessageWithCursor interface {
+	responseMessage
+	GetAfterResponseCursor() *v1.Cursor
+}
+
+type receiver[S responseMessage] interface {
+	Recv() (S, error)
+	grpc.ClientStream
+}
+
+const secondaryCursorPrefix = "$$secondary:"
+
+func publishClient[Q requestMessageWithCursor, R responseMessageWithCursor](ctx context.Context, client receiver[R], stream dispatch.Stream[R], secondaryDispatchName string) error {
```

Comment:
is `Q` used anywhere?

Reply:
Its used for the type of the incoming message, off of which I grab the cursor. I could have the req just be that interface directly, but I always prefer well-typed wherever possible